diff --git a/src/Data/List/Split/Internals.hs b/src/Data/List/Split/Internals.hs
index 03c3431..5452e3a 100644
--- a/src/Data/List/Split/Internals.hs
+++ b/src/Data/List/Split/Internals.hs
@@ -245,12 +245,7 @@ dropInitial _ l = l
 
 -- | Drop a final blank chunk according to the given 'EndPolicy'.
 dropFinal :: EndPolicy -> SplitList a -> SplitList a
-dropFinal _ [] = []
-dropFinal DropBlank l = dropFinal' l
- where
-  dropFinal' [] = []
-  dropFinal' [Text []] = []
-  dropFinal' (x : xs) = x : dropFinal' xs
+dropFinal DropBlank l | Text [] <- last l = init l
 dropFinal _ l = l
 
 -- * Combinators
