diff --git a/src/Data/List/Split/Internals.hs b/src/Data/List/Split/Internals.hs
index 810c7a0..03c3431 100644
--- a/src/Data/List/Split/Internals.hs
+++ b/src/Data/List/Split/Internals.hs
@@ -529,7 +529,7 @@ endByOneOf = split . dropFinalBlank . dropDelims . oneOf
 
 {-# DEPRECATED unintercalate "Use splitOn." #-}
 unintercalate :: (Eq a) => [a] -> [a] -> [[a]]
-unintercalate = endBy
+unintercalate = splitOn
 
 -- | Split into \"words\", with word boundaries indicated by the given
 --   predicate.  Satisfies @'Data.List.words' === wordsBy
